src/helpers/amazon-q/amazon-q-helpers.ts

```diff
@@ -1,86 +1,31 @@
-import { Credentials } from 'aws-sdk';
-import { ExpiredTokenException } from '@aws-sdk/client-sts';
+import { ChatSyncCommandOutput, SourceAttribution, MetadataEvent } from '@aws-sdk/client-qbusiness';
 
 const logger = makeLogger('amazon-q-helpers');
 
-// Member must have length less than or equal to 7000
-const AMAZON_Q_MSG_LIMIT = 7000;
-const WARN_TRUNCATED = `| Please note that you do not have all the conversation history due to limitation`;
 
-export const chat = async (
-  slackUserId: string,
-  incomingMessage: string,
-  attachments: AttachmentInput[],
-  dependencies: ChatDependencies,
-  env: SlackEventsEnv,
-  iamSessionCreds: Credentials,
-  context?: {
-    conversationId: string;
-    parentMessageId: string;
-  }
-): Promise<ChatSyncCommandOutput | Error> => {
-  try {
-    // Enforce max input message limit - may cause undesired side effects
-    // TODO consider 'smarter' truncating of number of chat history messages, etc. rather
-    // than simple input string truncation which may corrupt JSON formatting of message history
-    const inputMessage =
-      incomingMessage.length > AMAZON_Q_MSG_LIMIT
-        ? incomingMessage.slice(
-            incomingMessage.length + WARN_TRUNCATED.length - AMAZON_Q_MSG_LIMIT
-          ) + WARN_TRUNCATED
-        : incomingMessage;
-
-    const response = await dependencies.callClient(
-      slackUserId,
-      inputMessage,
-      attachments,
-      env,
-      iamSessionCreds,
-      context
-    );
-    logger.debug(`AmazonQ chatSync response: ${JSON.stringify(response)}`);
-    return response;
-  } catch (error) {
-    logger.error(`Caught Exception: ${JSON.stringify(error)}`);
-    if (error instanceof Error) {
-      logger.debug(error.stack);
-      if (error instanceof ExpiredTokenException) {
-        logger.error(`Token expired: ${error.message}`);
-      }
-      return new Error(error.message);
-    } else {
-      return new Error(`${JSON.stringify(error)}`);
-    }
-  }
-};
-
-export const getResponseAsBlocks = (response: ChatSyncCommandOutput) => {
-  if (isEmpty(response.systemMessage)) {
+export const getResponseAsBlocks = (content: string, systemMessageId: string, sourceAttributions: SourceAttribution[]) => {
+  if (isEmpty(content)) {
     return [];
   }
 
-  const content = response.systemMessage;
-
   return [
     ...(!hasTable(content)
       ? getMarkdownBlocks(convertHN(content))
       : getMarkdownBlocks(
           `${convertHN(getTablePrefix(content))}\n\n${parseTable(getTable(content))}`
         )),
-    ...(!isEmpty(response.sourceAttributions)
-      ? [createButton('View source(s)', response.systemMessageId ?? '')]
+    ...(!isEmpty(sourceAttributions)
+      ? [createButton('View source(s)', systemMessageId ?? '')]
       : [])
   ];
 };
 
-export const getFeedbackBlocks = (response: ChatSyncCommandOutput): Block[] => [
+
+export const getFeedbackBlocks = (response: ChatSyncCommandOutput | MetadataEvent): Block[] => [
```

Comment:
how come the response can be a MetadataEvent? Not sure what that type is

Reply:
Ack. I was trying to avoid duplicate logic i.e. two functions here - I can split it into 2, 1 for ChatSync & Chat respectively to avoid the conditional logic.